homework/list.js

```diff
@@ -1,79 +1,98 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+    var TodoListManager = function () { //constructor
+        this._listNoteContent = [];
+        this._wrapper = document.querySelector('#note-list-wrapper');
+    }
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+    TodoListManager.prototype = { //its prototype
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+        start() {
+             new Promise(function (resolve) {
+                 this.fetchList(function(data){
```

Comment:
It's better to design `fetchList` as a function return Promise, so there is no need to use callback style.

For example:

    this.fetchList()
      .then((function() {
        this.updateList(data);
        this.drawList();
        this.preloadFirstNote();
      }).bind(this))

It also clear the code since there is no need to wrap another Promise on that.